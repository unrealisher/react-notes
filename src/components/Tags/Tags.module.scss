.list {
  display: flex;
  flex-wrap: wrap;
}

.tag {
  font-size: 14px;
  line-height: 22px;
  padding: 0 8px;
  border-radius: 16px;
  background-color: rgba(51, 51, 51, 0.1);
  margin-right: 12px;
  margin-bottom: 12px;
}
