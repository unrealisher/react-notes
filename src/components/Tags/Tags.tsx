import React from "react";

import styles from "./Tags.module.scss";

interface IProps {
  tags?: string[];
}

export const Tags = (props: IProps): JSX.Element => {
  const getTag = (tag: string): string => {
    if (tag.length > 33) {
      return tag.substr(0, 30) + "...";
    }
    return tag;
  };

  const { tags } = props;
  return (
    <ul className={styles.list}>
      {tags &&
        tags.map(tag => {
          return (
            <li key={tag} className={styles.tag}>
              {getTag(tag)}
            </li>
          );
        })}
    </ul>
  );
};

export default Tags;
