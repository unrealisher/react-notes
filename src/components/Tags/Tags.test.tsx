import React from "react";
import { shallow } from "enzyme";
import { Tags } from "./Tags";

describe("Tags", () => {
  const tags = ["Покупки", "Работа", "ШРИ"];
  it("correct render", () => {
    const component = shallow(<Tags tags={tags} />);
    expect(component).toMatchSnapshot();
  });
});
