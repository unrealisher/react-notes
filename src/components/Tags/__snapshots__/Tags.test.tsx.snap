// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tags
    tags={
      Array [
        "Покупки",
        "Работа",
        "ШРИ",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="tag"
        >
          Покупки
        </li>,
        <li
          className="tag"
        >
          Работа
        </li>,
        <li
          className="tag"
        >
          ШРИ
        </li>,
      ],
      "className": "list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Покупки",
        "nodeType": "host",
        "props": Object {
          "children": "Покупки",
          "className": "tag",
        },
        "ref": null,
        "rendered": "Покупки",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "Работа",
        "nodeType": "host",
        "props": Object {
          "children": "Работа",
          "className": "tag",
        },
        "ref": null,
        "rendered": "Работа",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "ШРИ",
        "nodeType": "host",
        "props": Object {
          "children": "ШРИ",
          "className": "tag",
        },
        "ref": null,
        "rendered": "ШРИ",
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="tag"
          >
            Покупки
          </li>,
          <li
            className="tag"
          >
            Работа
          </li>,
          <li
            className="tag"
          >
            ШРИ
          </li>,
        ],
        "className": "list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Покупки",
          "nodeType": "host",
          "props": Object {
            "children": "Покупки",
            "className": "tag",
          },
          "ref": null,
          "rendered": "Покупки",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "Работа",
          "nodeType": "host",
          "props": Object {
            "children": "Работа",
            "className": "tag",
          },
          "ref": null,
          "rendered": "Работа",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "ШРИ",
          "nodeType": "host",
          "props": Object {
            "children": "ШРИ",
            "className": "tag",
          },
          "ref": null,
          "rendered": "ШРИ",
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
